ui/src/validation.ts

```diff
@@ -48,8 +52,20 @@ interface FormatConstraints {
     pattern: RegExp;
     message: string;
   };
+  firstHandleChar?: firstHandleCharOptions;
```

Comment:
Alternatively, you could skip making a separate `interface` and just do:
```suggestion
  firstHandleChar?: {
      valueName: string
  };
```